docs/data/toolpad/core/components/dashboard-layout/dashboard-layout.md

```diff
@@ -126,6 +126,12 @@ The layout sidebar can be hidden if needed with the `hideNavigation` prop.
 
 {{"demo": "DashboardLayoutSidebarHidden.js", "height": 400, "iframe": true}}
 
+### Toggle sidebar
```

Comment:
This section needs to be a bit more complete as in referring to the whole concept of using "controlled state" for the navigation menu, as well as what kind of use case they could serve.

We also need to document the new callbacks, and how they could be used together with `navigationMenuOpen`.

Anyway, I can take care of the documentation part when the functionality is done unless you really want to tackle it!